.github/workflows/push-commit-to-pull-request.yml

```diff
@@ -1,180 +1,53 @@
 ---
-name: e2e-tests-main-devnet
+#  This workflow is intended to run only for PRs on main and release branches,
+#  and does following
+#  * it builds test an aleph-node and runtime, wraps it in Docker,
+#  * in parallel, it builds production node and runtime, and check runtime determinism
+#  * it builds an e2e test client, wraps it in Docker,
+#  * in parallel, it starts an aleph-node consensus and run an e2e test on it
+#  * also in parallel, it checks clippy, unit tests and compiles excluded packages
+#  The whole workflow passes only if all e2e tests pass.
+name: PR
```

Comment:
I think we could name it more descriptive than just "PR".

Reply:
Please see now, concept changed a bit